spec/unit/indirector/rest_spec.rb

```diff
@@ -723,6 +723,49 @@ def save_request(key, instance, options={})
     end
   end
 
+  describe '#handle_response' do
+    # There are multiple request types to choose from, this may not be the one I want for this situation
+    let(:response) { mock_response(200, 'body') }
+    let(:connection) { stub('mock http connection', :put => response, :verify_callback= => nil) }
+    let(:instance) { model.new('the thing', 'some contents') }
+    let(:request) { save_request(instance.name, instance) }
+
+    before :each do
+      terminus.stubs(:network).returns(connection)
+    end
+
+    it 'adds server_agent_version to the context if not already set' do
+      Puppet.expects(:push_context).with({:server_agent_version => Puppet.version})
+      terminus.handle_response(request, response)
+    end
+
+    it 'does not add server_agent_version to the context if it is already set' do
+      Puppet.push_context({:server_agent_version => "5.3.4"})
+
+      Puppet.expects(:push_context).never
+      terminus.handle_response(request, response)
```

Comment:
Does the spec_helper or lib/puppet/test/helper ensure we rollback so that the stack isn't unbalanced? Might be better to handle it yourself:

```
Puppet.push_context(:server_agent_version => "5.3.4") do
  Puppet.expects(:push_context).never
  terminus.handle_response(request, response)
end
```